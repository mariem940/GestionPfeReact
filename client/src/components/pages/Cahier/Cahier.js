import React, { useState , useEffect} from 'react';
import './style-cahier.css';
import { Link } from 'react-router-dom'
import { useHistory, Redirect } from "react-router-dom"

function Cahier ({id, title, description, definition, objectifs, consignes, techniques, deleteTask, updateTask}) {
  const [updateMode, setUpdateMode] = useState(false)
  const [titleToUpdate, setTitleToUpdate] = useState(title)
  const [descriptionToUpdate, setDescriptionToUpdate] = useState(description)
  const [defintionToUpdate, setDefinitionToUpdate] = useState(definition)
  const [objectifsToUpdate, setObjectifsToUpdate] = useState(objectifs)
  const [consignesToUpdate, setConsignesToUpdate] = useState(consignes)
  const [techniquesToUpdate, setTechniquesToUpdate] = useState(techniques)
  const handleUpdateTask = () => {
    updateTask(id, titleToUpdate, descriptionToUpdate, defintionToUpdate, objectifsToUpdate, consignesToUpdate, techniquesToUpdate)
    setUpdateMode(false)
  }
  console.log({id}) 
 const history = useHistory()
 const handleDetails = () => {
   history.push(`/tasks/${id}`)
 }
    return (
      
        <div className="container">
           {/* {taskId && <Redirect to={`/tasks/${id}`} /> } */}
          {!updateMode ? (
             <>
             {/* <div onClick={handleDetails} className="link"> */}
            <div class="card">
            <Link to={`/tasks/${id}`}>
            <h5 class="card-header">{title}</h5>
            </Link>
          {/* </div> */}
          
   
    <div class="col">
    <a href="#" class="btn btn-primary" onClick={() => deleteTask(id)}>Delete</a>
    </div>
    <br></br>
    <div class="col">
    <a href="#" class="btn btn-primary" onClick={() => setUpdateMode(true)}>Update</a>
    </div>
        </div>
        
        </>
          ) : (
            <div className="container">
      <div className="py-4">
        <div>
          <div class="form-group">
            <h2>Remplir votre cahier</h2>
        <label htmlFor="titre">Titre de projet</label>
          <input type="text"  name="title" value={titleToUpdate} onChange={(e) => setTitleToUpdate(e.target.value)} placeholder="écrivez le titre de votre projet" className="formField" />
          <label htmlFor="description">La description générale du projet</label>
          <input type="text"  name="description" value={descriptionToUpdate} onChange={(e) => setDescriptionToUpdate(e.target.value)} placeholder="écrivez la description de votre projet" className="formField" />
          <label htmlFor="definition">La définition globale du projet</label>
          <input type="text"  name="definition" value={defintionToUpdate} onChange={(e) => setDefinitionToUpdate(e.target.value)} placeholder="écrivez la définition globale de votre projet" className="formField" />
          <label htmlFor="objectifs">Les objectifs</label>
          <input type="text"  name="objectifs" value={objectifsToUpdate} onChange={(e) => setObjectifsToUpdate(e.target.value)} placeholder="écrivez les objectifs de votre projet" className="formField" />
          <label htmlFor="consignes">Les consignes d'acquisition</label>
          <input type="text"  name="consignes" value={consignesToUpdate} onChange={(e) => setConsignesToUpdate(e.target.value)} placeholder="écrivez les consignes d'acquisition de votre projet" className="formField" />
          <label htmlFor="techniques">Les spécifications techniques</label>
          <input type="text"  name="techniques" value={techniquesToUpdate} onChange={(e) => setTechniquesToUpdate(e.target.value)} placeholder="écrivez votre spécifications techniques" className="formField" />
          <button onClick={handleUpdateTask} className="myButton">Update </button>
        </div>
        </div>
        </div>
        </div>
          )}
        </div>
        
     
    );
  }


export default Cahier;
