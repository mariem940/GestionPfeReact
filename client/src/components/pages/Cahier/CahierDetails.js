import React, { useEffect, useState } from "react"
import store from '../../../data/store';
import axios from "axios"
import { connect } from 'react-redux';
import { toast } from 'react-toastify';
import { fetchTaskById } from "../../services/tasks.service"
import { useParams, useLocation } from "react-router-dom"
import {addValidation as addValidationFromService} from '../../services/pfe.service'


import { Redirect } from 'react-router-dom';



function CahierDetails({enseignant, isAuth, isLoading}) {
  const [loading, setLoading] = useState(false)
  const [task, setTask] = useState({})
  const [validation, setValidations] = useState([])

  
 
 //const { userId } = useParams()
  const { taskId } = useParams()
  
  
 
  console.log('useParams(): ', useParams());
  console.log('useLocation(): ', useLocation());

  useEffect(() => {
    const fetchData = async () => {
      setLoading(true)
      const result = await fetchTaskById(taskId)
      setTask(result)
      setLoading(false)
    }
    fetchData()
  }, [taskId])


  const handleAddValidation = () => {
    if (enseignant) {
      const {_id} = enseignant
      console.log(_id)
    
    
    addValidation(taskId, _id) 
    console.log(taskId, _id)
  }
  }

  const addValidation = async (cahier_id, user_id) => {
    try {
    
      const newValidation = await addValidationFromService({
       cahier_id, user_id
        
      })
      setValidations([...validation, newValidation])
    
    } catch (err) {
      const errors = err.response.data.errors
        if (errors) {
            errors.forEach(error => toast.error(error.msg))
        }

    }
  }
 
  

  return (
    <div className="task-details">
     
      {loading ? (
        <div>Loading ... </div>
      ) : (
        <>
      <div className="container">
      <br></br>
    <div class="card">
  <h5 class="card-header">{task.title}</h5>
  <div class="card-body">
    <h5 class="card-title">La description générale du projet</h5>
    <p class="card-text">{task.description}</p>
    <h5 class="card-title">La définition globale du projet</h5>
    <p class="card-text">{task.definition}</p>
    <h5 class="card-title">Les objectifs</h5>
    <p class="card-text">{task.objectifs}</p>
    <h5 class="card-title">Les consignes d'acquisition</h5>
    <p class="card-text">{task.consignes}</p>
    <h5 class="card-title">Les spécifications techniques</h5>
    <p class="card-text">{task.techniques}</p>
  

    <div class="col">
     
    <button onClick={handleAddValidation} className="myButton">Validation </button>
    </div>
    
  </div>
</div>
</div>
        </>
      )}
    </div>
  )
}

const mapToStateProps = (state) => ({
  isAuth: state.authprof.isAuthenticated,
  isLoading: state.authprof.loading,
  enseignant: state.authprof.enseignant,
});
export default connect(mapToStateProps, )(CahierDetails);



