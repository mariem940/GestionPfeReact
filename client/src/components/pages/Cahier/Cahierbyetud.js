import React, { useState, useEffect } from "react";

import axios from "axios";
import { connect } from 'react-redux';



const CahierByEtud = (etudiant, isAuth, isLoading) => {
  const [cahiers, setCahier] = useState([]);
 
  
 
 useEffect(() => {
   
    loadCahierByEtud();
 }, []);


 if (etudiant.etudiant) {
  const {_id} = etudiant.etudiant
  console.log(etudiant.etudiant)}


  const loadCahierByEtud = async (_id) => {
    if (etudiant.etudiant) {
      const {_id} = etudiant.etudiant
      console.log(etudiant.etudiant)
      console.log(_id)
    
    const result = await axios.get(`http://localhost:4000/api/cahier/`+_id);
    setCahier(result.data);
    }
  };
 

 
  return (
    <div className="container">
      <br></br>
    <div className="card">
    <div className="card-body">
    {cahiers.map((cahier, index) => (

      <div>
        key={cahier.id}
            id={cahier._id}
  <h5 className="card-header">{cahier.title}</h5>

    <h5 className="card-title">La description générale du projet</h5>
    <p className="card-text">{cahier.description}</p>
    <h5 className="card-title">La définition globale du projet</h5>
    <p className="card-text">{cahier.definition}</p>
    <h5 className="card-title">Les objectifs</h5>
    <p className="card-text">{cahier.objectifs}</p>
    <h5 className="card-title">Les consignes d'acquisition</h5>
    <p className="card-text">{cahier.consignes}</p>
    <h5 className="card-title">Les spécifications techniques</h5>
    <p className="card-text">{cahier.techniques}</p>
    <h5 className="card-title">Publier par</h5>
    <p className="card-text">{cahier.user_id.nomutilisateur}</p>
    <br></br>
 </div>
   ))}
   </div>
   </div>
</div>

  );
};



const mapToStateProps = (state) => ({
  isAuth: state.authetud.isAuthenticated,
  isLoading: state.authetud.loading,
  etudiant: state.authetud.etudiant,
});
export default connect(mapToStateProps, )(CahierByEtud);