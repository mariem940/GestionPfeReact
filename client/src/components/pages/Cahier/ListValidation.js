import React, { useState, useEffect } from "react";
import { Link, useParams } from "react-router-dom";
import axios from "axios";
import {
  toast
} from 'react-toastify';

import { useLocation } from "react-router-dom"
import { connect } from 'react-redux';


const ListValidation = () => {

    const [validations, setValidation] = useState([])
  useEffect(() => {
    loadValidation();
  }, []);

 
  


  const loadValidation = async () => {
    const result = await axios.get(`http://localhost:4000/api/validation/all`);
    setValidation(result.data.reverse());

  };
  
 
  return (
    <div className="container">
      <br></br>
    <div className="card">
    <div className="card-body">
    {validations.map((validation, index) => (

      <div>
        key={validation.id}
            id={validation._id}
  <h5 className="card-header">{validation.cahier_id.title}</h5>

    <h5 className="card-title">Valid√© par</h5>
    <p className="card-text">{validation.user_id.nom}</p>
   
    
 </div>
   ))}
   </div>
   </div>
</div>

  );
};


export default ListValidation;


