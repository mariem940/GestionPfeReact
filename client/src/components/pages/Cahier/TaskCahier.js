import React, { useState } from "react"
import ViewCahier from "./ViewCahier"
import { connect } from 'react-redux';
function TaskCahier({ addTask, etudiant, isAuth, isLoading}) {
  const [title, setTitle] = useState("")
  const [description, setDescription] = useState("")
  const [definition, setDefinition] = useState("")
  const [objectifs, setObjectifs] = useState("")
  const [consignes, setConsignes] = useState("")
  const [techniques, setTechniques] = useState("")

  if (etudiant) {
    const {_id} = etudiant
    console.log(_id)}
    console.log(etudiant)
  const handleAddTask = () => {
    if (etudiant) {
      const {_id} = etudiant
      console.log(_id)
    addTask(title, description, definition, objectifs, consignes, techniques, _id)
    setTitle("")
    setDescription("")
    setDefinition("")
    setObjectifs("")
    setConsignes("")
    setTechniques("")
  }
}

  return (
    <div className="container">
      <div className="py-4">
        <div>
          <div class="form-group">
            <h2>Remplir votre cahier</h2>
        <label htmlFor="titre">Titre de projet</label>
          <input type="text"  name="title" value={title} onChange={(e) => setTitle(e.target.value)} placeholder="écrivez le titre de votre projet" className="formField" />
          <label htmlFor="description">La description générale du projet</label>
          <input type="text"  name="description" value={description} onChange={(e) => setDescription(e.target.value)} placeholder="écrivez la description de votre projet" className="formField" />
          <label htmlFor="definition">La définition globale du projet</label>
          <input type="text"  name="definition" value={definition} onChange={(e) => setDefinition(e.target.value)} placeholder="écrivez la définition globale de votre projet" className="formField" />
          <label htmlFor="objectifs">Les objectifs</label>
          <input type="text"  name="objectifs" value={objectifs} onChange={(e) => setObjectifs(e.target.value)} placeholder="écrivez les objectifs de votre projet" className="formField" />
          <label htmlFor="consignes">Les consignes d'acquisition</label>
          <input type="text"  name="consignes" value={consignes} onChange={(e) => setConsignes(e.target.value)} placeholder="écrivez les consignes d'acquisition de votre projet" className="formField" />
          <label htmlFor="techniques">Les spécifications techniques</label>
          <input type="text"  name="techniques" value={techniques} onChange={(e) => setTechniques(e.target.value)} placeholder="écrivez votre spécifications techniques" className="formField" />
          <button onClick={handleAddTask} className="myButton">submit </button>
        </div>
        </div>
        </div>
        </div>
  )
}


const mapToStateProps = (state) => ({
  isAuth: state.authetud.isAuthenticated,
  isLoading: state.authetud.loading,
  etudiant: state.authetud.etudiant,
});
export default connect(mapToStateProps, )(TaskCahier);

