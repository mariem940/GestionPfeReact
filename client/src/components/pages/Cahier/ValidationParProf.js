import React, { useState, useEffect } from "react";
import { Link, useParams } from "react-router-dom";
import axios from "axios";
import {
  toast
} from 'react-toastify';

import { useLocation } from "react-router-dom"
import { connect } from 'react-redux';
import { fetchValidationsParProf } from "../../services/pfe.service";


const ValidationParProf = (enseignant, isAuth, isLoding) => {

    const [validations, setValidation] = useState([])
 useEffect(() => {
  
    loadValidationParProf();
  }, [])
;

  if ( enseignant.enseignant) {
    const { _id } = enseignant.enseignant;
    console.log(enseignant.enseignant)
console.log(_id)
  }

  const loadValidationParProf = async(_id) => {
    if ( enseignant.enseignant) {
      const { _id } = enseignant.enseignant
      console.log(enseignant.enseignant)
      console.log(_id)
      const result = await axios.get(`http://localhost:4000/api/validation/enseignant/`+_id);
    console.log(result.data)
    setValidation(result.data)
    console.log(validations)
    }
  
    
  };

 
  return (
    <div className="container">
      <br></br>
    <div className="card">
    <div className="card-body">
    {validations.map((validation, index) => (

      <div>
        key={validation.id}
            id={validation._id}
  <h5 className="card-header">{validation.cahier_id.title}</h5>

    <h5 className="card-title">Valid√© par</h5>
    <p className="card-text">{validation.user_id.nom}</p>
   
    
 </div>
   ))}
   </div>
   </div>
</div>

  );
};

const mapToStateProps = (state) => ({
   
    isLoading: state.authprof.loading,
    enseignant: state.authprof.enseignant,
  });
  export default connect(mapToStateProps, )(ValidationParProf);



