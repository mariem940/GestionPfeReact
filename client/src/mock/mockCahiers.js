export default [
    {
        id: "",
        title: "",
        description: "",
        definition:"",
        objectifs:"",
        consignes:"",
        techniques:""
    },
    
]