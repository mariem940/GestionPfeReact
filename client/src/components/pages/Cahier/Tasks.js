import React from "react"
import Cahier from "./Cahier"


export default function TasksList({ tasks, deleteTask, updateTask }) {
  return (
    <div className="tasks-list">
      <div>
        {tasks.map((task) => (
          <Cahier
            key={task.id}
            id={task._id}
            title={task.title}
            description={task.description}
            definition={task.definition}
            objectifs={task.objectifs}
            consignes={task.consignes}
            techniques={task.techniques}
            user_id={task.user_id.nomutilisateur}
            deleteTask={deleteTask}
            updateTask={updateTask}
           
          />
        ))}
      </div>
    </div>
  )
}
