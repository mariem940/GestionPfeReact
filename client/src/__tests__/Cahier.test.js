import React from "react"
import { render } from "@testing-library/react"
//import userEvent from "@testing-library/user-event"
import mockCahiers from "../mock/mockCahiers"
import Cahier from "../components/pages/Cahier/Cahier";
describe("test cahiers list", () => {
    test("should render an array of cahiers list", () => {
        const mockDeleteTask = jest.fn()
        // const CahiersComponent = render (
        //     <Cahier deleteTask={mockDeleteTask} tasks={mockCahiers}/>

        //)
        expect().toMatchSnapshot()
        //debug()
    })
})