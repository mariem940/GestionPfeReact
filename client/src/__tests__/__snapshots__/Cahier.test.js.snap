// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test cahiers list should render an array of cahiers list 1`] = `undefined`;
